mapiproxy/libmapiproxy/dcesrv_mapiproxy_session.c

```diff
@@ -40,63 +44,54 @@
    \return Pointer to an allocated mpm_session structure on success,
    otherwise NULL
  */
-struct mpm_session *mpm_session_new(TALLOC_CTX *mem_ctx, 
-				     struct server_id serverid,
-				     uint32_t context_id)
+struct mpm_session *mpm_session_new(struct server_id serverid,
+				    uint32_t context_id,
+				    const char *username,
+				    struct GUID	*uuid)
 {
 	struct mpm_session	*session = NULL;
 
-	if (!mem_ctx) return NULL;
-
-	session = talloc_zero(mem_ctx, struct mpm_session);
+	session = talloc_zero(NULL, struct mpm_session);
 	if (!session) return NULL;
 
 	session->server_id = serverid;
 	session->context_id = context_id;
-	session->ref_count = 0;
+	if (uuid) session->uuid = *uuid;
```

Comment:
It'd be safer to create a copy and avoid taking the reference pointer. It's the same case as username field below